pkg/registry/apis/secret/contracts/inline.go

```diff
@@ -0,0 +1,19 @@
+package contracts
+
+import (
+	"context"
+
+	common "github.com/grafana/grafana/pkg/apimachinery/apis/common/v0alpha1"
+)
+
+type InlineSecureValueSupport interface {
+	// Check that the request user can reference a secret in the context of a given resource (owner)
+	CanReference(ctx context.Context, owner common.ObjectReference, values common.InlineSecureValues) error
+
+	// CreateInline creates a secret that is owned by the referenced object
+	// returns the name of the created secret or an error
+	CreateInline(ctx context.Context, owner common.ObjectReference, value common.RawSecureValue) (string, error)
+
+	// DeleteInline removes secrets that are owned by a referenced object
+	DeleteInline(ctx context.Context, owner common.ObjectReference, name string) error
```

Comment:
```suggestion
	// DeleteIfInline removes secrets that are owned by a referenced object
	DeleteIfInline(ctx context.Context, owner common.ObjectReference, name string) error
```

Not sure the right word... but this should delete IFF the saved secret is actually owned by the object.  From the client, we do not know the difference.  It should not be an error if we do not own it; rather just a noop

Reply:
`DeleteWhenOwnedByResource`? maybe